substrate/frame/nomination-pools/src/lib.rs

```diff
@@ -2663,8 +2689,9 @@ pub mod pallet {
 		#[pallet::weight(T::WeightInfo::chill())]
 		pub fn chill(origin: OriginFor<T>, pool_id: PoolId) -> DispatchResult {
 			let who = ensure_signed(origin)?;
-
 			let bonded_pool = BondedPool::<T>::get(pool_id).ok_or(Error::<T>::PoolNotFound)?;
+			// ensure pool is not in an un-migrated state.
+			ensure!(!Self::api_pool_needs_delegate_migration(pool_id), Error::<T>::NotMigrated);
```

Comment:
How about not gating this extrinsic with the need for migration? Instead of migrating, the pool owner could just chill and destroy it.

Reply:
You cannot destroy pool as that would mean kicking existing members including the initial depositor which can only work once they are migrated.

But practically, all pools will get migrated. There are only some members who are also staking that will be blocked. So this is only defensive.